### PR TITLE
python3Packages.astroid: 2.4.2 -> 2.5.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

Python 3.9 :heart: 

Updates to 2.5.0 which was released today (2020-02-16).
https://github.com/PyCQA/astroid/releases/tag/astroid-2.5

<s>Applies patches that fix Python 3.9 support in the testsuite:
- https://github.com/PyCQA/astroid/pull/829
- https://github.com/PyCQA/astroid/pull/858
</s>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/astroid/default.nix [hunk 1]
```diff
@@ -23,7 +56,9 @@ buildPythonPackage rec {
     ++ lib.optional (pythonOlder "3.5") typing
     ++ lib.optional (!isPyPy) typed-ast;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/astroid/default.nix:57]: I think we do not need typed-ast for 3.9 https://github.com/PyCQA/astroid/blob/astroid-2.4.2/astroid/__pkginfo__.py#L32

Comment (mweinelt) [reply pkgs/development/python-modules/astroid/default.nix:57]: In fact we only need it < 3.8.

**Comment by mweinelt** [comment]
Meanwhile a new upstream release happened and that simplifies the situation alot.

File: pkgs/development/python-modules/astroid/default.nix [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
-    substituteInPlace astroid/__pkginfo__.py --replace "lazy_object_proxy==1.4.*" "lazy_object_proxy"
-  '';
-
   # From astroid/__pkginfo__.py
-  propagatedBuildInputs = [ lazy-object-proxy six wrapt ]
-    ++ lib.optional (pythonOlder "3.5") typing
-    ++ lib.optional (!isPyPy) typed-ast;
+  propagatedBuildInputs = [
+    lazy-object-proxy
+    six
```
Comment (misuzu) [inline pkgs/development/python-modules/astroid/default.nix:30]: `six` was removed from dependencies.

Comment (mweinelt) [reply pkgs/development/python-modules/astroid/default.nix]: Removed. Thanks!

Comment (misuzu) [reply pkgs/development/python-modules/astroid/default.nix]: You forgot to remove `six` from function arguments on line 9

Comment (mweinelt) [reply pkgs/development/python-modules/astroid/default.nix]: I should stop doing many things in parallel. I clearly loose track of things. :roll_eyes:

**Review by misuzu — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112908` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>home-assistant: log was empty</li>
<li>hydra-unstable: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.astroid</li>
<li>python38Packages.astroid</li>
<li>python39Packages.astroid</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
hydra-unstable:
<details>


`gpl3` is a deprecated license, check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/hydra/common.nix:133:5:

```
    |
133 |     license = licenses.gpl3;
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
autoconf is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
automake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
perl is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
subversion is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
unzip is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/development/tools/misc/hydra/common.nix:81:3:

```
   |
81 |   buildInputs =
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md
Unused argument: `fetchFromGitHub`.
Near pkgs/development/tools/misc/hydra/common.nix:1:40:

```
  |
1 | { stdenv, nix, perlPackages, buildEnv, fetchFromGitHub
  |                                        ^
```
Unused argument: `darcs`.
Near pkgs/development/tools/misc/hydra/common.nix:3:23:

```
  |
3 | , top-git, mercurial, darcs, subversion, breezy, openssl, bzip2, libxslt
  |                       ^
```
Unused argument: `guile`.
Near pkgs/development/tools/misc/hydra/common.nix:4:3:

```
  |
4 | , guile, perl, postgresql, nukeReferences, git, boehmgc, nlohmann_json
  |   ^
```
</details>
python37Packages.astroid:
<details>


`lgpl2` is a deprecated license, check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/astroid/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python38Packages.astroid:
<details>


`lgpl2` is a deprecated license, check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/astroid/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
python39Packages.astroid:
<details>


`lgpl2` is a deprecated license, check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/astroid/default.nix:36:5:

```
   |
36 |     license = licenses.lgpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by SuperSandro2000** [comment]
home-assistant and hydra-unstable fail because of pylint.
```
builder for '/nix/store/8wpmpz95mfjk0dni2615lbdc5zqg3x58-python3.8-pylint-2.6.2.drv' failed with exit code 1;
setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/pylint-2.6.2/dist /build/pylint-2.6.2
Processing ./pylint-2.6.2-py3-none-any.whl
Requirement already satisfied: isort<6,>=4.2.5 in /nix/store/s5zcphj9zjcq1pfywhd4jnhdg26m60n8-python3.8-isort-5.6.4/lib/python3.8/site-packages (from pylint==2.6.2) (5.6.4)
Requirement already satisfied: mccabe<0.7,>=0.6 in /nix/store/lxmvlrkj6iw53cvcizl13pj7kg1922n5-python3.8-mccabe-0.6.1/lib/python3.8/site-packages (from pylint==2.6.2) (0.6.1)
Requirement already satisfied: toml>=0.7.1 in /nix/store/pi4w43xlpjdgd3prhy6xajzxjv5a8lz2-python3.8-toml-0.10.2/lib/python3.8/site-packages (from pylint==2.6.2) (0.10.2)
ERROR: Could not find a version that satisfies the requirement astroid<2.5,>=2.4.0 (from pylint)                                                                                            ERROR: No matching distribution found for astroid<2.5,>=2.4.0
cannot build derivation '/nix/store/am7x7wanma6l6cgkw0mdnqc2jsk61wfg-python3.8-trio-0.17.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/7h0xd0v0qy1ln9y6hc3j16cyw1ik7bgc-python3.8-jeepney-0.6.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x99jr39mmpafrkq0a05sn789sph9lcb6-python3.8-pytest-trio-0.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/psq4mg96gk33f5wz1lx0akwla44r2m8g-python3.8-httpx-0.16.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/jag9m4p0hxwvfssrikfqwryh10hpmqyf-python3.8-secretstorage-3.3.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/y5wvr0c2c5c3kwy55cmf1w0g53j2wlzw-homeassistant-2021.2.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/x3cz8ml6dp7v8sm2xxjih5pz5p88zxpx-python3.8-keyring-21.5.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/rg7w78lvj4r05rygkqfk5ziv4pz5jb97-python3.8-launchpadlib-1.10.13.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/ip69k8g13svg3a9g0y69nnnijj09w8i8-python3.8-breezy-3.1.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/xlracj47dm51wa56q4pl9vn47lx67mji-hydra-2020-10-20.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/jza91fhl9dnnhxyfvz39nbi51kngrzqz-review-shell.drv': 2 dependencies couldn't be built
```

**Comment by mweinelt** [comment]
The latest pylint release (2.6.2, 2 days ago) is apparently not yet compatible with astroid 2.5.0.

**Comment by misuzu** [comment]
pylint 2.7.1 was released yesterday: https://github.com/PyCQA/pylint/releases/tag/pylint-2.7.1

**Comment by mweinelt** [comment]
@FRidh This is good to go.

**Comment by bobrik** [comment]
#116814 fixes darwin build of `pylint` after this PR.

**Pull request merged**